src/Form/FormAnalyzer.js

```diff
@@ -293,7 +307,8 @@ class FormAnalyzer {
 
         // Check form contents (noisy elements are skipped with the safeUniversalSelector)
         const selector = this.matching.cssSelector('safeUniversalSelector');
-        const formElements = findEnclosedElements(this.form, selector);
+        const elements = this.form.querySelectorAll(selector);
+        const formElements = elements.length > 0 ? [...elements] : findElementsInShadowTree(this.form, selector);
```

Comment:
Seems like we used to have this nicely encapsulated in `findEnclosedElements`. Why did we remove that nice abstraction and now have to repeat these `querySelectorAll` calls explicitly?